build.gradle.kts

```diff
@@ -195,8 +195,20 @@ tasks {
 			if (modularBuild.toBoolean())
 				java.srcDir("src/main/module")
 		}
+		create("demo") {
```

Comment:
can we maybe utilize the test suite feature for this?

Reply:
```
testing {
  suites {
    val test by getting(JvmTestSuite::class) { useJUnitJupiter() }

    val demoTests by
        registering(JvmTestSuite::class) {
          dependencies { implementation(project) }

          sources { java { srcDir("src/demo/java") } }
          targets { all { testTask.configure { shouldRunAfter(test) } } }
        }
  }
}
```

would do the trick, but we would need to update our gradle wrapper to version 7.4, and apply the `jvm-test-suite` plugin - with that a demoTest task will be automatically added.

The dependson section would look like
```
dependsOn(javadoc, validateYaml, testing.suites.named("demoTests"))
```

Reply:
What's the advantage of that over the current configuration?

Reply:
it is more concise and you do not need to manuially configure the configruations and the classpaths. also the test tasks is automatically created, and the whole configuration for `demotests` is just the block

Reply:
Okay, I think we should involve the others and discuss on the 2022-02-10 maintainer meeting (because of the expanded scope).

Reply:
i am not sure what you mean with expanded scope - this is just a new feature which was specially introduced by gradle for such a case - before that, you had to do it, like it is now :)

Reply:
I thought the gradle update would be considered expanded scope... Now I'm not so sure.

Reply:
okay yeah the upgrade from 7.3 to 7.3.3 might be a good topic. and i agree

Reply:
I don't see a reason to discuss. Updating a build-time dependency to the next minor/patch version is everyday business. 😉

Reply:
Oh, and I think @aepfli's proposal to use test suites is 100% worth the update.